block_stats.go

```diff
@@ -0,0 +1,278 @@
+package seth
+
+import (
+	"context"
+	"fmt"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pelletier/go-toml/v2"
+	"go.uber.org/ratelimit"
+	"golang.org/x/sync/errgroup"
+	"math"
+	"math/big"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+type BlockStatsConfig struct {
```

Comment:
is it possible to make it generic to be used for ethclient?

Reply:
It is possible but I don't think you'll use it in an automated way, it's just a quick analysis CLI.